lab-michaelporter/index.js

```diff
@@ -0,0 +1,97 @@
+'use strict';
+
+const net = require('net');
+const server = net.createServer();
+const Client = require('./lib/client.js');
+
+let clientPool = [];
+
+server.on('connection', (socket) => {
+  let client = new Client(socket);
+
+  clientPool = [...clientPool, client];
+
+  clientPool.forEach(item => {
+    item.socket.write(`Welcome ${client.nick}\n`);
+  });
+
+  let disconnect = (err) => {
+    if (err) return console.error(err);
+    clientPool = clientPool.filter((item) => item.socket !== client.socket);
+    clientPool.forEach((item) => {
+      item.socket.write(`Bye ${client.nick}\n`);
+    });
+  };
+
+  socket.on('error', disconnect);
+  socket.on('close', disconnect);
+  socket.on('data', (buffer) => {
+    let data = buffer.toString();
+    console.log(data);
+
+    if (data.startsWith('/nick')) {
+      let oldNick = client.nick;
+      client.nick = data.split(' ')[1] || client.nick;
+      client.nick = client.nick.trim();
+      if (client.nick === oldNick || client.nick === '') {
+        client.nick = oldNick;
+        socket.write(`${client.nick} Change your nick by typing "/nick newNick"\n`);
+        return;
+      }
+      clientPool.forEach((item) => {
+        item.socket.write(`${oldNick} is now ${client.nick}\n`);
+      });
+      return;
+    }
+
+    if (data.startsWith('/dm')) {
+      let splitData = data.split(' ') || '';
+      let contact = splitData[1];
+      // if (contact === '' || clientPool.indexOf(contact) === -1) {
```

Comment:
so you're currently trying to check if 'contact' has an index in the clientPool array. You actually aren't being quite specific enough here. 'contact' is only supposed to be representing the client nick(nickname), right? Because of that, we need to search for the index of 'client.contact' instead of just 'contact'